neo/test/controllers/ApplicationTest.scala

```diff
@@ -0,0 +1,25 @@
+package controllers
+
+import java.io.File
+
+import org.specs2.mutable.Specification
+
+/**
+  * Created by zongh on 10/26/2016.
+  */
+class ApplicationTest extends Specification {
+
+  "application" should {
+    "load the city data from a file" in {
+      val cities = Application.loadCity(new File("public/data/city.json"))
+      cities.size must_== 29834
+    }
+
+    "calculate centroid" in {
+      val cities = Application.loadCity(new File("public/data/city.sample.json"))
```

Comment:
to make test faster, I think we should only use the `city.sample.json` in the test.
and we should move `public/data/city.sample.json` to `test/resource/data/city.sample.json` and only access by `/data/city.sample.json`.